compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -644,14 +644,15 @@ trait Applications extends Compatibility {
               missingArg(n)
           }
 
-          if (formal.isRepeatedParam)
+          val formal1 = formal.stripFlexible
```

Comment:
I have a bit of a concern that we are really  not sure where else FlexibleTypes would appear and cause missing cases. Why do a `stripFlexible` here? For the following test it should ideally be irrelevant because `a repeated parameter` should never be a flexible type. Can we achieve that as an invariant? On the other hand, further down we do `formal1.widenExpr.argTypesLo`. What if the flexible type is a by-name type? Does not happen from Java but what when old Scala code is also treated with flexible types? Would it not be better to not do a `stripFlexible` here but make sure `argTypesLo` will also work for flexible types?

My fear is that we will get a long tail of errors because we are not sure where we need to strip flexible types.

Reply:
Given the fact that varargs itself can be nullable, I agree it would be better to handle flexible types inside `isRepeatedParam`, `argTypesLo`, and other related functions.

Reply:
`typeSymbol` and `argInfos` should work for flexible types now, so no need to strip Null or flexible types before checking repeated params now.